client/components/data/domain-management/index.jsx

```diff
@@ -35,66 +35,62 @@ function getStateFromStores( props ) {
 	};
 }
 
-const DomainManagementData = createReactClass( {
-	displayName: 'DomainManagementData',
-
-	propTypes: {
+class DomainManagementData extends React.Component {
+	static propTypes = {
 		analyticsPath: PropTypes.string,
 		analyticsTitle: PropTypes.string,
 		context: PropTypes.object.isRequired,
-		productsList: PropTypes.object.isRequired,
 		selectedDomainName: PropTypes.string,
 		selectedSite: PropTypes.object,
 		sitePlans: PropTypes.object.isRequired,
-	},
-
-	mixins: [ observe( 'productsList' ) ],
+	};
 
-	componentWillMount: function() {
+	componentWillMount() {
 		const { selectedSite } = this.props;
 
 		if ( selectedSite ) {
 			fetchDomains( selectedSite.ID );
 		}
-	},
+	}
 
-	componentWillUpdate: function( nextProps ) {
+	componentWillUpdate( nextProps ) {
 		const { selectedSite: prevSite } = this.props;
 		const { selectedSite: nextSite } = nextProps;
 
 		if ( nextSite && nextSite !== prevSite ) {
 			fetchDomains( nextSite.ID );
 		}
-	},
+	}
 
-	render: function() {
+	render() {
 		return (
-			<div>
+			<Fragment>
 				<PageViewTracker path={ this.props.analyticsPath } title={ this.props.analyticsTitle } />
+				<QueryProducts />
```

Comment:
should we gate these on whether or not we have a valid products list? Or do we want to requery on every render when we're not already fetching?

Reply:
The Flux store only ever requests the data from network once per app lifetime. And persists them to local storage to make sure they are available immediately on app reboot.

I'll update `QueryProducts` to behave the same way. I believe the developer should only need to declare `<QueryProducts />` and let the implementation be smart enough to know when to issue a network request and when not.